src/plugins/data/public/search/session/session_service.ts

```diff
@@ -215,6 +230,18 @@ export class SessionService {
    * Cleans up current state
    */
   public clear() {
+    // make sure apps can't clear other apps' sessions
+    const currentSessionApp = this.state.get().appName;
+    if (currentSessionApp && currentSessionApp !== this.currentApp) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        `Skip clearing session "${this.getSessionId()}" because it belongs to a different app. current: "${
+          this.currentApp
+        }", owner: "${currentSessionApp}"`
+      );
+      return;
```

Comment:
Not sure if I agree with this. 
Because currently you still can "clear" others' app session by starting a new session, right?

Reply:
Didn't you add this code yourself in 
https://github.com/elastic/kibana/pull/91874/files#diff-278ed05f7a3365f7178b74d7b100f82cfb07eef4e9ada0e087013700336d5b8eR209 
?

Reply:
Oh, I see, thanks for finding the change! 

But I am missing now: was it removed and now you're adding it back? Is this right? Why?